mcpserver/tools.go

```diff
@@ -340,55 +365,60 @@ Supported schemas are:
+
 		config := env.Config.Copy()
+		var suggestions []string
 
-		instructions, err := request.RequireString("instructions")
-		if err != nil {
-			return nil, err
+		// Build specific cu config commands based on what was requested
+		if baseImage != "" {
+			config.BaseImage = baseImage
+			suggestions = append(suggestions, fmt.Sprintf("cu config base-image set %s", baseImage))
 		}
-		config.Instructions = instructions
 
-		baseImage, err := request.RequireString("base_image")
-		if err != nil {
-			return nil, err
+		if len(setupCommands) > 0 {
+			config.SetupCommands = setupCommands
+			for _, cmd := range setupCommands {
+				suggestions = append(suggestions, fmt.Sprintf("cu config setup-command add \"%s\"", cmd))
+			}
 		}
-		config.BaseImage = baseImage
 
-		setupCommands, err := request.RequireStringSlice("setup_commands")
-		if err != nil {
-			return nil, err
+		if err := env.UpdateConfig(ctx, config); err != nil {
+			return mcp.NewToolResultErrorFromErr("unable to update the environment", err), nil
 		}
-		config.SetupCommands = setupCommands
 
-		envs, err := request.RequireStringSlice("envs")
-		if err != nil {
-			return nil, err
+		if err := repo.Update(ctx, env, request.GetString("explanation", "")); err != nil {
+			return mcp.NewToolResultErrorFromErr("failed to update repository", err), err
 		}
-		config.Env = envs
 
-		secrets, err := request.RequireStringSlice("secrets")
-		if err != nil {
-			return nil, err
-		}
-		config.Secrets = secrets
+		// Create directive message for the agent
+		message := `SUCCESS: Configuration successfully applied. Environment has been restarted, all previous commands have been lost.
+The changes are temporary, you MUST inform the user to make the changes persist
 
-		if title := request.GetString("title", ""); title != "" {
-			env.State.Title = title
-		}
+`
 
-		if err := env.UpdateConfig(ctx, request.GetString("explanation", ""), config); err != nil {
-			return mcp.NewToolResultErrorFromErr("unable to update the environment", err), nil
+		if explanation != "" {
+			message += fmt.Sprintf("TELL THE USER: I need the following environment configuration: %s\n\n", explanation)
 		}
 
-		if err := repo.Update(ctx, env, request.GetString("explanation", "")); err != nil {
-			return mcp.NewToolResultErrorFromErr("unable to update the environment", err), nil
+		if len(suggestions) > 0 {
+			message += "YOU MUST tell the user to run these commands to make the environment configuration changes:\n\n"
+			for _, suggestion := range suggestions {
+				message += fmt.Sprintf("  %s\n", suggestion)
+			}
+			message += "\nExplain to the user that after running these commands, future environments will use the updated configuration."
+		} else {
+			message += "If environment configuration changes are needed, inform the user they can run:\n\n"
+			message += "  cu config base-image set <image>     # Set base container image\n"
+			message += "  cu config setup-command add \"<cmd>\"  # Add setup command\n"
```

Comment:
this probably needs to do a diff of new and existing setup commands?

Reply:
Yeah the prompt is very broken-ish, not optimized, just toying with the idea